docs/2.guide/5.best-practices/security.md

```diff
@@ -0,0 +1,306 @@
+
+* Works on both the client (Nuxt) and the server (Nitro)
+* Write abilities once and use them everywhere
+* Agnostic of the authentication layer
+* Use components to conditionally show part of the UI
+* Primitives are can be accessed for a full customization
+
+Abilities constitute the rules that embody the `authorization logic`. They take a user and one or more resources and yield a deny or an allow condition. While they can be grouped by resource, they remain independent of one another.
+
+```ts
+export const editBook = defineAbility((user: User, book: Book) => {
+  return user.id === book.authorId
+})
+```
+
+The `defineAbility` function acts as a factory that creates an ability but nothing beyond that. It needs to be used with bouncer functions. In a server endpoint, the `authorize` function can be employed to grant access to a resource based on the abilities
+
+```ts
+export default defineEventHandler(async (event) => {
+  await authorize(event, editBook)
+
+  const books = await db.query.books.findMany()
+
+  return books
+})
+```
+
+:read-more{title="Nuxt Authorization" to="https://soubiran.dev/posts/nuxt-going-full-stack-how-to-handle-authorization"}
+
+
+## Common problems
+
+When building more complex Nuxt applications, you will probably encounter some of the problems listed below. Understanding these problems and fixing them will help make your website more secure.
+
+### Returning entities with sensitive data
+
+**Problem**: When fetching the data from external source (i.e. Database) and returning it in the Nuxt API endpoint, you may return the whole table entity (such as User) to the browser and handle it there. This can lead to unintentional leaking of sensitive information that may not even be needed. 
+
+**Solution**: Utilize `pick` or use `interceptors` concepts for both `useAsyncData` and `useFetch` to sanitize what you get from the server without leaking confidential data.
+
+```ts
+const { data } = await useFetch('/api/modules', {
+  pick: ['title']
+})
+```
+
+:read-more{title="useFetch" to="/docs/api/composables/use-fetch#usage"}
+
+### Handling State with Care in Nuxt SSR
+
+**Problem**: Using `ref` in Nuxt SSR applications can lead to unintentionally share the state between requests, leading to potential data leaks and inconsistent application behavior.
+
+```ts
+const unsafeGlobal = ref<number>(1);
+
+export function useSafeRef() {
+  const safeRef = ref<number>(2);
+
+  return {
+    unsafeGlobal,
+    safeRef
+  }
+}
+```
+
+:read-more{title="CrossRequestStatePollution" to="https://vuejs.org/guide/scaling-up/ssr.html#cross-request-state-pollution"}
+
+**Solution**: For enhanced safety and to adhere to Nuxt's best practices, it's recommended to manage state using abstractions with `useState`:
+
+```ts
+const safeGlobal = useState<number>('safeGlobal', () => 5);
+
+export function useSafeRef() {
+  const safeRef = useState<number>('safeRef', () => 2);
+
+  return {
+    safeGlobal,
+    safeRef
+  }
+}
+```
```

Comment:
that's not true. Having useState in a global var will make all request using this file fail because `useState` is called at file import -> nuxt will not be available

Reply:
Are you talking about sharing something among all SSR requests ? 
If so, i'd use a nitro plugin to inject into the event. But i'm always against the usage of objects in the global scope anyway. It"s way too easy to cause a memleak.

For global state management, within a single NuxtApp. There's a few solutions. 
Use `useState` in a composable. UseState is a state within the NuxtApp. 
Use a NuxtPlugin to provide data
Use Pinia (which has a plugin then provide composables)